scripts/build.sh

```diff
@@ -14,6 +14,35 @@ log() {
     printf "\033[${color}m%s:\033[0m %s\n" "$level" "$message" >&2
 }
 
+check_ccache_usage() {
+    # Check if ccache is full and warn user
+    cache_stats=$(ccache --print-stats)
+    current_kb=$(echo "$cache_stats" | grep "^cache_size_kibibyte" | cut -f2)
+    max_kb=$(echo "$cache_stats" | grep "^max_cache_size_kibibyte" | cut -f2)
+
+    if [ -n "$current_kb" ] && [ -n "$max_kb" ] && [ "$max_kb" -gt 0 ] 2>/dev/null; then
+        usage_percent=$((current_kb * 100 / max_kb))
+
+        if [ "$usage_percent" -gt 90 ] 2>/dev/null; then
+            # Convert to human-readable sizes (GiB)
+            current_gb=$((current_kb / 1024 / 1024))
+            max_gb=$((max_kb / 1024 / 1024))
+            log warning "ccache is ${usage_percent}% full (${current_gb}/${max_gb} GiB). Consider increasing cache size with 'ccache -M <size>G'"
```

Comment:
Since this gets piped into printf, better escape the `%`:
```suggestion
            log warning "ccache is ${usage_percent}%% full (${current_gb}/${max_gb} GiB). Consider increasing cache size with 'ccache -M <size>G'"
```

Reply:
Does it really need escaping if it's not the printf format string? Testing that change I see a double '%%' being printed

Reply:
Oh apologies, I didn't realize it was getting substituted as an `%s` argument: good the way you wrote it.